android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/shopping/GeckoShoppingProductAnalysis.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.shopping
+
+import mozilla.components.concept.engine.shopping.ShoppingProductAnalysis
+
+/**
+ * Holds the result of the analysis of a shopping product.
+ *
+ * @property productId Product identifier (ASIN/SKU)
+ * @property analysisURL Analysis URL
+ * @property grade Reliability grade for the product's reviews
+ * @property adjustedRating Product rating adjusted to exclude untrusted reviews
+ * @property needsAnalysis Boolean indicating if the analysis is stale
+ * @property lastAnalysisTime Time since the last analysis was performed
+ * @property deletedProductReported Boolean indicating if reported that this product has been deleted
+ * @property deletedProduct Boolean indicating if this product is now deleted
+ * @property highlights Object containing highlights for product
+ */
+data class GeckoShoppingProductAnalysis(
+    override val productId: String,
+    val analysisURL: String?,
+    val grade: String?,
```

Comment:
think we should use an enum here

Reply:
oh, I did not update you on the enum story.

So initially I agreed with you that since they use a limited set of values (American school grades), this can totally be an enum. But then I decided to make sure, just in case - I asked Jared if it's possible that their API is going to change in the future. His answer indicated that there is, in fact, such possibility (although not in MVP, but still).

So I decided that the initial design with String is more robust. Because if they decide to change the set of values, or make then numbers etc - we would have to change throughout the whole stack, and that would involve deprecating an API - which takes 2 release cycles in GV. I would much rather have this more vague design that doesn't have the potential to generate tech debt down the road.

When using this API in Fenix or Focus, you can make this an enum of `{"A", "B", "C", "D", "F"}` - but again, keep in mind the API is not stable yet, and we are not sure if any other characters (such as "?") would be or are used, etc.

Reply:
It's a tech debt because we would have to spend time updating GV API (writing a new one, deprecating old one, removing dead code), and AC. This time could be spent more productively. I think an enum in Fenix layer is fine - in that case there would be only one layer to change should the API change.

>  It should never return ? but null in that case IMO.

Hey, I did not design that API :) I'm with you on this one :) but it is what it is. I am referring to the Gecko API. (not GeckoView API)